.github/workflows/lint.yaml

```diff
@@ -50,3 +50,20 @@ jobs:
         run: python -m pip install gitlint
       - name: Run gitlint check
         run: gitlint --commits origin/${{ github.event.pull_request.base.ref }}..HEAD
+  checkton:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Run Checkton
+        id: checkton
+        uses: chmeliik/checkton@v0.1.2
+        with:
```

Comment:
The original [PR](https://github.com/konflux-ci/build-definitions/pull/1051/files) also sets `find-copies-harder: true`. Did we not want that?

More importantly, the acceptance criteria in Jira say that we should fix all the issues that the tool finds. But by default it uses differential mode, so only new code is checked. So we don't know if there are issues in the existing code.

Reply:
The Checkton tool is designed to check only new code by default. In a  Konflux demo call, the owner demonstrated this feature and explained that this decision was made because linting the entire codebase would result in too many issues. If we want to lint the entire codebase, we would need to use a different tool along with Checkton.

`find-copies-harder: true`. Did we not want that?  We can add that if you would like ?

Reply:
Added in the  `find-copies-harder: true`

Reply:
> In a Konflux demo call, the owner demonstrated this feature and explained that this decision was made because linting the entire codebase would result in too many issues.

OK, but in our jira we specifically said we did want to fix all the issues it finds in our current code. And it can be accomplished by setting differential mode to false - it could be a separate PR where you temporarily set that to see what needs fixing, fix all the issues and eventually remove the differential=false. I think it would be worth to at least see how many issues it would find and then we can decide if we want to fix all of those or not.

Reply:
Got it, I see what you mean now. Here's the link to the PR with the linted code problems: https://github.com/seanconroy2021/app-interface-deployments/pull/2. There are only a few issues, so I think it makes sense to include them here.

Reply:
Cool, thanks. But why does that PR show 23 commits? Either you'll need to update your fork's main, but you could also create that PR against the main repo's main. And keep it as Draft (and perhaps with some "Test" prefix) so that it's not accidentally merged.

And I agree - it actually seems it's just the one thing repeated in several places?
````sh
export KRB5_CONFIG=`mktemp`
```
I'm not exactly sure what the warning is saying though. I would start by changing it to `export KRB5_CONFIG=$(mktemp)` and maybe that will fix it.

So you can add a second commit to this PR with that fix (and reference that testing PR so that anybody can check and see why those things are being fixed).